src/java.base/share/classes/java/lang/reflect/code/type/JavaType.java

```diff
@@ -218,17 +209,11 @@ static JavaType type(JavaType t, JavaType... typeArguments) {
     }
 
     static JavaType type(JavaType t, List<JavaType> typeArguments) {
-        if (t.isPrimitive()) {
-            throw new IllegalArgumentException("Cannot parameterize a primitive type");
-        } else if (t.isArray()) {
-            return array(type(((ArrayType)t).componentType(), typeArguments));
-        } else {
-            ClassType ct = (ClassType)t;
-            if (ct.hasTypeArguments()) {
-                throw new IllegalArgumentException("Type must not have type arguments: " + ct);
-            }
-            return new ClassType(ct.toClassName(), typeArguments);
-        }
+        return switch (t) {
```

Comment:
Again, the old code was not complete - in the `else` branch we assumed the only possible option was a class type, which is no longer true